modules/expo-bluesky-translate/android/src/main/java/expo/modules/blueskytranslate/ExpoBlueskyTranslateModule.kt

```diff
@@ -0,0 +1,65 @@
+package expo.modules.blueskytranslate
+
+import com.google.mlkit.common.model.DownloadConditions
+import com.google.mlkit.nl.translate.Translation
+import com.google.mlkit.nl.translate.TranslatorOptions
+import expo.modules.kotlin.modules.Module
+import expo.modules.kotlin.modules.ModuleDefinition
+import expo.modules.kotlin.Promise
+
+class ExpoBlueskyTranslateModule : Module() {
+    var translationPromise: Promise? = null
+
+    override fun definition() = ModuleDefinition {
+        Name("ExpoBlueskyTranslate")
+
+        AsyncFunction("translateAsync") { sourceLanguage: String, targetLanguage: String, text: String, promise: Promise ->
+            if (translationPromise != null) {
+                promise.reject(TranslationInProgressException())
+                return@AsyncFunction
+            }
+
+            translationPromise = promise
+
+            val options = TranslatorOptions.Builder()
+                .setSourceLanguage(sourceLanguage)
+                .setTargetLanguage(targetLanguage)
+                .build()
+
+            val translator = Translation.getClient(options)
+
+            val conditions = DownloadConditions.Builder()
+                .requireWifi()
```

Comment:
models are around 30MB, so the documentation advises to use WiFi before downloading the models